index.js

```diff
@@ -847,6 +955,303 @@ function flip(message, side, coinage) {
+                            //decrease buyer quantity by amount sold, remove seller posting
+                            buyers[Bkey].quantity -= sellers[Skey].quantity;
+                            delete sellers[Skey];
+                        }
+                        //Selling > Buying
+                        else if (sellers[Skey].quantity > buyers[Bkey].quantity){
+                            var buyerData = ledger[Bkey];
+                            var sellerData = ledger[Skey];
+                            //credit both parties as they have already paid
+                            buyerData.meatCoin += parseFloat(buyers[Bkey].quantity);
+                            volume.bought += parseFloat(buyers[Bkey].quantity);
+                            sellerData.gold += buyers[Bkey].quantity * negPrice;
+                            if (buyerData.history.length > 9)
+                                buyerData.history.pop();
+                            buyerData.history.unshift('b' + ',' + parseFloat(buyers[Bkey].quantity).toFixed(2) + ',' + negPrice.toFixed(2));
+                            if (sellerData.history.length > 9)
+                                sellerData.history.pop();
+                            sellerData.history.unshift('s' + ',' + parseFloat(buyers[Bkey].quantity).toFixed(2) + ',' + negPrice.toFixed(2));
+                            //decrease seller quantity by amount sold, remove buyer posting
+                            sellers[Skey].quantity -= buyers[Bkey].quantity;
+                            delete buyers[Bkey];
+                        }
+                        //equal
+                        else{
+                            var buyerData = ledger[Bkey];
+                            var sellerData = ledger[Skey];
+                            //credit both parties as they have already paid
+                            buyerData.meatCoin += parseFloat(buyers[Bkey].quantity);
+                            volume.bought += parseFloat(buyers[Bkey].quantity);
+                            sellerData.gold += buyers[Bkey].quantity * negPrice;
+                            if (buyerData.history.length > 9)
+                                buyerData.history.pop();
+                            buyerData.history.unshift('b' + ',' + parseFloat(buyers[Bkey].quantity).toFixed(2) + ',' + negPrice.toFixed(2));
+                            if (sellerData.history.length > 9)
+                                sellerData.history.pop();
+                            sellerData.history.unshift('s' + ',' + parseFloat(buyers[Bkey].quantity).toFixed(2) + ',' + negPrice.toFixed(2));
+                            //decrease seller quantity by amount sold, remove buyer posting
+                            delete sellers[Skey];
+                            delete buyers[Bkey];
+                        }
+                    }
+                })
+            }
+        });
+    }
+}
+
```

Comment:
Try to space out the code. Put a newline before all the comments (thank you for having comments) and some of the if statements.